src/main/resources/templates/go/solution/mergesort.go

```diff
@@ -0,0 +1,57 @@
+package ${packageName}
+
+import "time"
+
+type MergeSort struct{}
+
+func NewMergeSort() *MergeSort {
+	return new(MergeSort)
+}
+
+// PerformSort is a wrapper method for the real MergeSort algorithm.
+func (m MergeSort) PerformSort(input []time.Time) {
+	mergeSort(input)
+}
+
+// mergeSort recursively applies the MergeSort algorithm.
+func mergeSort(input []time.Time) {
```

Comment:
If you wanted to align it closer to the java code, you'd also need to add the receiver types here

Reply:
I don't think we should follow Java as closely here. In Java these functions are class members by necessity. They are more something like utility functions used by the sorting method. Recursively passing the receiver would also have an impact on performance.